### PR TITLE
Add support for Pay-As-You-Go (PAYG) CSP billing adapter

### PR DESCRIPTION
**Pull request by guangyee**
Also check for the `rancher-csp-billing-adapter` for PAYG offering.

### Summary
This is a feature enhancement to support Pay-As-You-Go offering for Public Cloud.

### Occurred changes and/or fixed issues
This change simply checking for the additional application to support PAYG offering.

### Technical notes summary
This change is to enable the support config and support theme when the PAYG application is detected.

### Areas or cases that should be tested
We need to make sure both Bring-Your-Own (BYO) and PAYG offering supported, and that this change will not break the original BYO offering.

### Areas which could experience regressions
The existing BYO offering.

### CODE REVIEW

**Review by nwmac — APPROVED** [review]

**Comment by nwmac** [comment]
@guangyee Remind me what release this needs to go in for? @jeremy-moffitt ?

**Comment by jeremy-moffitt** [comment]
We're targeting the release after 2.7.5 ... per discussion in Slack I understand this may be 2.7.6 or 2.7.7... The ideal would be to get this into the Q3 release rather than waiting for 2.8 since we only need this and 1 other change in the Rancher codebase for the project we're pushing, and landing those for 2.7.6 would mean we would be our own "long pole" in the schedule rather than the Rancher release being the schedule driver.

**Review by richard-cox — DISMISSED** [review]
I pushed an update to make the search through apps more efficient. While doing that i discovered a scenario (possible regression) where the ui-brand was not being set on fresh installs (see commit for more detail). 

If @nwmac is still happy i'll merge

Edit: Change introduced merge conflicts. I've only merged and not rebased to make the changes clearer, so we'll squash merge this back to `master`

**Review by nwmac — DISMISSED** [review]

**Comment by guangyee** [comment]
NOTE: new changes are related this conversation thread in Rancher. https://github.com/rancher/rancher/pull/41849#discussion_r1238844552

File: shell/pages/support/index.vue
Comment (jeremy-moffitt) [inline shell/pages/support/index.vue:103]: is this intentionally not setting a value for the parameter? is the backend just checking for the presence of the parameter rather than its value?

Comment (jeremy-moffitt) [reply shell/pages/support/index.vue:103]: I see in the corresponding backend change ( https://github.com/rancher/rancher/pull/41849/files#diff-5d7133457c7014b056d00275cc680aa1b50aa47b72b241be15dc4269b359496bR65 )  you're checking for the presence of the parameter, so that matches up... if the dashboard team is ok with that it works for me :)

Comment (MbolotSuse) [reply shell/pages/support/index.vue:103]: I had a similar comment on the backend side - `usePAYG=true` would read more cleanly to me, but I'm ok with it if the UI is (and we can find an RFC clarifying that single value parameters like this are ok).

Comment (richard-cox) [reply shell/pages/support/index.vue:103]: We're fine UI side. We don't do anything other than stick it in an anchor.  

The URI / HTML specs are a bit vague/absent about a missing value -https://www.rfc-editor.org/rfc/rfc3986#section-3.4.

Generally i think it's avoided. If there's no clear spec it probably gets treated differently by different backends (empty string/null/omitted/etc). If the place this is used handles it ok, i'd say we're good.

**Comment by richard-cox** [comment]
There were some exceptions thrown during the setup screens that were causing e2e jobs to fail. This was due to a fix i'd added to handle some ui-brand setting updates. I've now made that process safer.

**Review by richard-cox — DISMISSED** [review]

**Comment by richard-cox** [comment]
There is a [commit](https://github.com/guangyee/dashboard/commit/d771f88b7dc27c28497e5c41c9e32d5fbd435aa9) on the branch that _should_ resolve some of the flakyness, however for some reason isn't showing up in the PR. Will give GH some time to catch up, if not resolve will try pushing an empty commit

**Comment by richard-cox** [comment]
Some of the changes i've introduced to fix the case where the initial branding wasn't applied have caused some general issues, not just with the e2e tests. 

I'm going to tackle those separably in a different PR then come back and update this one

**Comment by richard-cox** [comment]
https://github.com/rancher/dashboard/pull/9269 should fix the underlying problem of initialised ui-brand when it doesn't exist. @guangyee Would you be able to have a quick look and test with your environment? I'll get someone in team this side to give it a general code review as well

**Comment by guangyee** [comment]
> #9269 should fix the underlying problem of initialised ui-brand when it doesn't exist. @guangyee Would you be able to have a quick look and test with your environment? I'll get someone in team this side to give it a general code review as well

I am able to test this PR in conjunction with https://github.com/rancher/rancher/pull/41849 in my local dev env. I was able to verify that, with the new PAYG CSP adapter installed, the generate support config button is visible and the color scheme is green. Whereas if the CSP adapter is absent, the support config button is not visible and the color scheme is blue.

File: shell/pages/support/index.vue
Comment (guangyee) [reply shell/pages/support/index.vue]: I've updated the backend https://github.com/rancher/rancher/pull/41849 to explicitly look for `usePAYG=true` as well.

**Comment by guangyee** [comment]
> @guangyee Would you be able to validate #9269 with the previous csp adaptor? I'd like to confirm there's no regressions before bringing in the changes to this PR. This PR will then be much cleaner and shouldn't have any e2e issues

I ran into a merge conflict with #9269. But after reconcile the changes, I was able to test the branding theme with the original CSP Adapter as well.

**Comment by richard-cox** [comment]
@guangyee The ask is to review the functionality of https://github.com/rancher/dashboard/pull/9269 on it's own and in isolation. Once that PR looks good and is merged I'll update this PR such that this contains the fixes from 9269. This PR will then solely be about handling the new csp adapter app

**Comment by guangyee** [comment]
> @guangyee The ask is to review the functionality of #9269 on it's own and in isolation. Once that PR looks good and is merged I'll update this PR such that this contains the fixes from 9269. This PR will then solely be about handling the new csp adapter app

I am able to test #9269 on it's own isolation. I can see the correct theme (green logos, etc) when the original `rancher-csp-adapter` is installed. And the UI reverted back to the (blue theme) when the adapter is uninstalled.

**Comment by jeremy-moffitt** [comment]
@richard-cox @guangyee is there more testing or work to be done on this? I think Guang had mentioned another PR this might need to be rebased off of?

**Review by richard-cox — DISMISSED** [review]
We got there in the end! Everything's reverted apart from the core changes. @guangyee guangyee if this PR looks good to you we can merge

File: shell/pages/support/index.vue
Comment (guangyee) [inline shell/pages/support/index.vue:103]: Rancher is looking for `usePAYG=true`. I'll update.

**Comment by guangyee** [comment]
Tested the latest commit on my local test env and it's working as expected.

**Review by richard-cox — APPROVED** [review]
Just waiting on CI, then will merge

**Pull request merged**
